### PR TITLE
Added support for more parameters in mod.conf

### LINKED ISSUE #9395: Expand mod.conf with additional meta information (author, version, copyright, etc.)
## Problem

While mod.conf has seen minor improvement in recent years, in particular by #6243 to contain the mod dependencies and description, I believe there is still value in further expanding its scope with additional meta information about mods, particularly for use by the Content DB as well as to anticipate the future creation of game configuration and management tools.

## Solutions

My original proposal cited below entailed the addition of four more fields, aside from the existing `name`, `description`, and `depends`. 

https://forum.minetest.net/viewtopic.php?p=305622#p305622

For compatibility reasons `name` should really be `title`. I also suggest adding an `icon` field (that refers to an image be located in the `/textures` subdirectory), a `license` field (that is the name of the license, not the full text), and a `website` field (that is a URL to the forum fopic or GitHub repository). Here's a revised example:

```
name = my_farming
description = a great mod for useful farming tools
depends = default, farming
icon = my_farming_icon.png
title = My Farming Mod
version = 1.0 beta
website = https://www.example.com/
author = supermodder
license = MIT
copyright = Copyright (c) 2017, supermodder
```

Incidentally, I discovered that @rubenwardy had suggested something similar in #2201, even though that issue was closed and only a couple of fields were actually incorporated into the official documentation. So in some respects, I'd like to resurrect that discussion and see if we can settle on a truly futureproof "standard" of sorts for mod.conf.

### PR DESCRIPTION
**Pull request by arihant2math**
Add compact, short information about your PR for easier understanding:

Adds support for a website and license in mod.conf. Closes #9395

This PR is a Ready for Review

### CODE REVIEW

**Comment by arihant2math** [comment]
Waiting for #12926 before adding links to the license and website. I feel that linting the license isn't really needed for now.

**Comment by rubenwardy** [comment]
This should be added to game.conf and texture_pack.conf too, and be part of get_content_info and ContentSpec rather ModSpec

**Comment by arihant2math** [comment]
Moved it to `ContentSpec` only, this should work.

File: doc/lua_api.txt [hunk 1]
```diff
@@ -212,6 +215,8 @@ A `Settings` file that provides meta information about the mod.
 * `release`: Ignore this: Should only ever be set by ContentDB, as it is an
              internal ID used to track versions.
 * `title`: A human-readable title to address the mod.
+* `website`: The mod's website
+* `license`: The mods license, should be a SPDX identifiers
```
Comment (srifqi) [inline doc/lua_api.txt:219]:
```suggestion
* `website`: The mod's website. It must start with either "http://" or "https://"
             in order to be displayed.
* `license`: The mod's license. It should be an SPDX identifier if possible.
             See https://spdx.org/licenses/
```

**Comment by arihant2math** [comment]
Fixed merge conflicts (hopefully)

**Review by srifqi — CHANGES_REQUESTED** [review]
Please check the rebasing of this PR (do **not** merge branch) to current `master` branch. You can read more [here](https://git-scm.com/book/en/v2/Git-Branching-Rebasing).

The `lua_api.txt` has been converted to `lua_api.md`. The documentation for the Modding API now goes there.

-------

Also, can we make it uniform whether the website or the license that should go first? The order is different between files.

**Comment by arihant2math** [comment]
Rebase complete, reapplying changes to `lua_api.md`

**Comment by arihant2math** [comment]
ok, ported the changes.

**Review by srifqi — CHANGES_REQUESTED** [review]
Sorry, but please rebase this PR (do **not** merge branch) to current `master` branch. It should only consists of your commits (around 15 in this PR) to make reviewing easier. You can read more [here](https://git-scm.com/book/en/v2/Git-Branching-Rebasing).

**Comment by JamesClarke7283** [comment]
Why was this PR closed?

**Comment by Zughy** [comment]
@JamesClarke7283 author's gone

**Pull request closed without merging**